deploy/helm/kubecf/assets/operations/stacks.yaml

```
@@ -0,0 +1,178 @@
+
+# set lifecycle bundles
+{{- $lifecycle_bundles := dict }}
+{{- $_ := set $lifecycle_bundles "docker" "docker_app_lifecycle/docker_app_lifecycle.tgz" }}
+{{- range $stack_name := $.Values.install_stacks }}
+  {{- $_ := set $lifecycle_bundles (printf "buildpack/%s" $stack_name) "buildpack_app_lifecycle/buildpack_app_lifecycle.tgz" }}
+{{- end }}
+{{- include "_capi.setProperty" (list "diego.lifecycle_bundles" $lifecycle_bundles) }}
+
+{{- if not $.Values.features.eirini.enabled }}
+# set list of pre-loaded rootfses
+- type: replace
+  path: /instance_groups/name=diego-cell/jobs/name=rep/properties/diego/rep/preloaded_rootfses
+  value:
+  {{- range $stack_name := $.Values.install_stacks }}
+  - {{ $stack_name }}:/var/vcap/data/rep/{{ $stack_name }}/rootfs.tar
+  {{- end }}
+
+  {{- range $stack_name := $.Values.install_stacks }}
+- type: replace
+  path: /instance_groups/name=diego-cell/env?/bosh/agent/settings/disks/-
+  value:
+    volumeMount:
+      name: rep-data
+      mountPath: /var/vcap/data/rep
+    filters:
+      job_name: "{{ $stack_name }}-rootfs-setup"
+      process_name: "{{ $stack_name }}-rootfs-setup"
+  {{- end }}
+{{- end }}
+
+# set list of all buildpacks in install order
+{{- /* XXX Should this also use _capi.setProperty? */}}
+- type: replace
+  path: /instance_groups/name=api/jobs/name=cloud_controller_ng/properties/cc/install_buildpacks
+  value:
+{{- range $stack_name := $.Values.install_stacks }}
+  {{- $_ := include "_config.lookup" (list $ "stacks" $stack_name "install_buildpacks") }}
+  {{- range $buildpack_shortname := $.kubecf.retval }}
+    {{- $release_name := include "_config.lookup" (list $ "stacks" $stack_name "buildpacks" $buildpack_shortname) }}
+    {{- $version := include "_config.lookup" (list $ "releases" $release_name "version") }}
+    {{- $filename := printf "%s-buildpack-%s-v%s.zip" $buildpack_shortname $stack_name $version }}
+  - name: "{{ include "_config.lookup" (list $ "releases" $release_name "buildpack") }}_buildpack"
+    file: "/var/vcap/data/shared-packages/{{ $release_name }}/{{ $filename }}"
+  {{- end }}
+{{- end }}
+
+# create pre-rendering scripts
+{{- range $stack_name := $.Values.install_stacks }}
+  {{- if not $.Values.features.eirini.enabled }}
+- type: replace
+  path: /instance_groups/name=diego-cell/jobs/name={{ $stack_name }}-rootfs-setup/properties/quarks?/pre_render_scripts/jobs/-
+  value: |
+    #!/usr/bin/env bash
+    set -o errexit -o nounset
+
+    target="/var/vcap/all-releases/jobs-src/{{ $stack_name }}/{{ $stack_name }}-rootfs-setup/templates/pre-start"
+    sentinel="${target}.patch_sentinel"
+    if [[ -f "${sentinel}" ]]; then
+      if sha256sum --check "${sentinel}" ; then
+        echo "Patch already applied. Skipping"
+        exit 0
+      fi
+      echo "Sentinel mismatch, re-patching"
+    fi
+
+    # Use the ephemeral data directory for the rootfs
+    perl -p -i -e "s#\\\$ROOTFS_PACKAGE/rootfs#/var/vcap/data/rep/{{ $stack_name }}/rootfs#" "${target}"
```

```suggestion
    perl -p -i -e 's#\$ROOTFS_PACKAGE/rootfs#/var/vcap/data/rep/{{ $stack_name }}/rootfs#' "${target}"
```

I'm still torn between this and the old `patch` style changes…

---

> I'm still torn between this and the old `patch` style changes…

The reason for the change is that the old patches were different between `cflinuxfs3` and `sle15`. The lines directly following are different in the 2 pre-start scripts, so the same patch would not apply to both. Using the substitution is slightly more forgiving to changes in the pre-start script.

---

Quoting change applied in 0ab9ace